.werf/defines/version-map.tmpl

```diff
@@ -0,0 +1,37 @@
+# parses version_map.yml file and adds info about supported k8s versions
+{{- define "parse_version_map" }}
+  {{ $context := . }}
+  # Version Map
+  {{- $versionMap := dict }}
+  {{- range $_, $edition := $.Editions }}
+   {{- if $edition.versionMapFile }}
+     {{- $versionMap = (merge $versionMap ($.Files.Get $edition.versionMapFile | fromYaml) )}}
+   {{- end }}
+   {{- if eq $.Env $edition.name }}
+     {{- break -}}
+   {{- end }}
+  {{- end }}
+
+  {{- $_ := set $context "CandiVersionMap" $versionMap }}
+  # kubernetes versions list
+  {{- $kubernetesVersions := list }}
+  {{- $eolKubernetesVersions := list }}
+  {{- $previewKubernetesVersions := list }}
+  {{- $defaultKubernetesVersion := "" }}
+  {{- range $k, $v := $versionMap.k8s }}
+    {{- $kubernetesVersions = append $kubernetesVersions ($k | toString ) }}
+    {{- if eq $v.status "preview" }}
+      {{- $previewKubernetesVersions = append $previewKubernetesVersions ($k | toString ) }}
+    {{- end }}
+    {{- if eq $v.status "end-of-life" }}
+      {{- $eolKubernetesVersions = append $eolKubernetesVersions ($k | toString ) }}
+    {{- end }}
+    {{- if $v.default }}
+      {{- $defaultKubernetesVersion = $k | toString }}
+    {{- end }}
+  {{- end }}
+  {{ $_ := set $context "kubernetesVersions" $kubernetesVersions }}
+  {{ $_ := set $context "previewKubernetesVersions" $previewKubernetesVersions }}
+  {{ $_ := set $context "eolKubernetesVersions" $eolKubernetesVersions }}
+  {{ $_ := set $context "defaultKubernetesVersion" $defaultKubernetesVersion }}
```

Comment:
let to add limiter on defaultKubernetesVersion if we have multiple default version, fail with error https://masterminds.github.io/sprig/flow_control.html

k8s updater can to add multiple default wersions by mistake. And we can skip it on the review

Reply:
we can do it with additional counter variable

Reply:
yep, i will do

Reply:
done